### ISSUE TITLE
Output redirection to file not working properly

### ISSUE THREAD

**Issue by albandil** [open-mpi/ompi#9627]
I am using Open MPI from the 5.0.x development branch (every time with updated git submodules and re-run top-level `./autogen.pl`) on openSUSE Leap 15.3 with GCC 10.3.

Starting with the commit a0e8c685feb20a5af056261df33247efb973ab29 I observe wrong output redirection when using `--output filename=log`. Specifically, *before* that commit, the following program correctly writes ten numbers to each of the per-rank output files:
```c
#include <mpi.h>
#include <stdio.h>
#include <unistd.h>

int main (int argc, char *argv[])
{
    MPI_Init(&argc, &argv);

    for (int n = 1; n <= 10; n++)
    {
        sleep(1);
        printf("%d\n", n);
    }

    MPI_Finalize();

    return 0;
}
```
After the above-mentioned commit, however, the output logs get broken. During the execution, the content of the log files get periodically erased before each write. At the end of the execution the files sometimes contain just the last printed number (10) or even nothing at all. Sometimes, the log file for the rank 0 is missing altogether.

**Comment by awlauria** [comment]
Thanks for the report @albandil !

@gpaulsen @janjust I added this to our critical items list. This is basic functionality that should just work.

**Comment by rhc54** [comment]
This is fixed here: https://github.com/openpmix/openpmix/pull/2392

Won't show up until I post the v5.0 update PR - will point this at it for auto-close when I do.

**Comment by awlauria** [comment]
Awesome, thanks @rhc54 !

**Comment by awlauria** [comment]
https://github.com/open-mpi/ompi/pull/9825

Merged to v5.0.x in the above prrte/pmix submodule update. Closing.
